server/service/osquery_utils/queries.go

```diff
@@ -570,7 +570,7 @@ var mdmQueries = map[string]DetailQuery{
 		// > location at any time.
 		//
 		// [1]: https://developer.apple.com/documentation/devicemanagement/fderecoverykeyescrow
-		Query:            `SELECT to_base64(group_concat(line)) as filevault_key FROM file_lines WHERE path='/var/db/FileVaultPRK.dat'`,
+		Query:            `SELECT to_base64(group_concat(line, x'0a')) as filevault_key FROM file_lines WHERE path='/var/db/FileVaultPRK.dat'`,
```

Comment:
Just for my own understanding, this is because if the encryption key had a newline in it, it would be on two lines when selecting from `file_lines`, and the old code would put them back together with commas instead of newlines, right? The `group_concat` was never because we expected to find multiple keys in the file, it's basically to "rebuild" the file that got split into lines by osquery.

Reply:
yes, exactly! I don't like this approach I proposed a while back, I think we'll need to revisit the way we fetch the key, because I don't think there are any guarantees about the order in which the rows are concatenated here.